mantle/cmd/kolet/kolet.go

```diff
@@ -80,10 +97,106 @@ func registerTestMap(m map[string]*register.Test) {
 	}
 }
 
+// dispatchRunExtUnit returns true if unit completed successfully, false if
+// it's still running (or unit was terminated by SIGTERM)
+func dispatchRunExtUnit(unitname string, sdconn *systemddbus.Conn) (bool, error) {
+	props, err := sdconn.GetAllProperties(unitname)
+	if err != nil {
+		return false, errors.Wrapf(err, "listing unit properties")
+	}
+
+	result := props["Result"]
+	if result == "exit-code" {
+		return false, fmt.Errorf("Unit %s exited with code %d", unitname, props["ExecMainStatus"])
+	}
+
+	state := props["ActiveState"]
+	substate := props["SubState"]
+
+	switch state {
+	case "inactive":
+		fmt.Printf("Starting %s\n", unitname)
+		sdconn.StartUnit(unitname, "fail", nil)
+		return false, nil
+	case "active":
+		{
+			switch substate {
+			case "exited":
+				maincode := props["ExecMainCode"]
+				mainstatus := props["ExecMainStatus"]
+				switch maincode {
+				case CLD_EXITED:
+					if mainstatus == int32(0) {
+						return true, nil
+					} else {
+						// I don't think this can happen, we'd have exit-code above; but just
+						// for completeness
+						return false, fmt.Errorf("Unit %s failed with code %d", unitname, mainstatus)
+					}
+				case CLD_KILLED:
+					// SIGTERM; we explicitly allow that, expecting we're rebooting.
+					if mainstatus == 15 {
+						return false, nil
+					} else {
+						return true, fmt.Errorf("Unit %s killed by signal %d", unitname, mainstatus)
+					}
+				default:
+					return false, fmt.Errorf("Unit %s had unhandled code %d", unitname, maincode)
+				}
+			case "running":
+				return false, nil
+			case "failed":
+				return true, fmt.Errorf("Unit %s in substate 'failed'", unitname)
+			default:
+				// Pass through other states
+				return false, nil
+			}
+		}
+	default:
+		return false, fmt.Errorf("Unhandled systemd unit state %s", state)
+	}
+}
+
+func runExtUnit(cmd *cobra.Command, args []string) error {
+	unitname := args[0]
+	if strings.Index(unitname, ".") < 0 {
```

Comment:
Would this fail in the case where a binary already has a `.` in it?

Reply:
Yep will fix in a bit

Reply:
This doesn't look fixed? I think we can just restrict it to service files (i.e. just dumbly check that it ends with `.service` and append it if not).